homeassistant/components/livisi/entity.py

```diff
@@ -0,0 +1,88 @@
+"""Code to handle a Livisi switches."""
+from __future__ import annotations
+
+from typing import Any
+
+from collections.abc import Mapping
+from aiolivisi.const import CAPABILITY_MAP
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import LivisiDataUpdateCoordinator
+
+from .const import DOMAIN, LIVISI_REACHABILITY_CHANGE
+
+
+class LivisiEntity(CoordinatorEntity[LivisiDataUpdateCoordinator], Entity):
+    """Represents a base livisi entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        device: dict[str, Any],
+        *,
+        use_room_as_device_name: bool = False,
+    ) -> None:
+        """Initialize the common properties of a Livisi device."""
+        self.config_details: Mapping[str, Any] = device["config"]
+
+        self.aio_livisi = coordinator.aiolivisi
+        self.config_entry = config_entry
```

Comment:
There is no need to hold a reference to `config_entry`. It is only used here inside `__init__`.
```suggestion
```